src/mpl/include/mpl_thread_argobots.h

```diff
@@ -50,9 +50,9 @@ typedef void (*MPL_thread_func_t) (void *data);
 void MPL_thread_create(MPL_thread_func_t func, void *data, MPL_thread_id_t * idp, int *errp);
 
 #define MPL_thread_exit()
-#define MPL_thread_self(id_) ABT_thread_self_id(id_)
-#define MPL_thread_join(id_) ABT_thread_join(id_)
-#define MPL_thread_same(id1_, id2_, same_)  ABT_thread_equal(id1_, id2_, same_)
+#define MPL_thread_self(idp_) ABT_thread_self(idp_)
+#define MPL_thread_join(idp_) ABT_thread_free(idp_)
+#define MPL_thread_same(idp1_, idp2_, same_)  ABT_thread_equal(*idp1_, *idp2_, same_)
```

Comment:
Are these corresponding to a recent argobots API changes?

Reply:
Thanks for taking a look at this PR!

The join change (`MPL_thread_join()`)  was made by this commit: https://github.com/pmodels/mpich/commit/5d3ec442461f576768c6cfae943a1d8808208cfe
In Argobots, join takes a thread descriptor, not a thread ID. In the Argobots case, a thread descriptor can be used as an ID, so I replaced it.

Reply:
So argobots have both "thread id" and "thread descriptor"? Can they be converted back and forth?

Reply:
Descriptor -> ID, yes.
ID -> Descriptor, no.

A thread descriptor is essentially a pointer, so two different threads can have the same value when their lifetimes are not overlapped.

In this use case, `thread descriptor` works well.

Reply:
I see. Thanks for the explanation.